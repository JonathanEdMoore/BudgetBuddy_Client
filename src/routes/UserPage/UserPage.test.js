import React from 'react'
import ReactDOM from 'react-dom'
import UserPage from './UserPage'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('UserPage smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><UserPage /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('UserPage snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<UserPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})