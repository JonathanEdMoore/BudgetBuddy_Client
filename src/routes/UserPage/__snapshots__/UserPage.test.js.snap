// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage snapshot test renders the UI as expected 1`] = `
<div>
  <main
    role="main"
  >
    <header>
      <MainHeader />
      <p>
        Your Balance: $
         
      </p>
    </header>
    <label
      htmlFor="month"
    >
      Month 
    </label>
    <select
      name="month"
      onChange={[Function]}
    >
      <option
        value=""
      >
        Select Month
      </option>
      <option
        value="01"
      >
        January
      </option>
      <option
        value="02"
      >
        February
      </option>
      <option
        value="03"
      >
        March
      </option>
      <option
        value="04"
      >
        April
      </option>
      <option
        value="05"
      >
        May
      </option>
      <option
        value="06"
      >
        June
      </option>
      <option
        value="07"
      >
        July
      </option>
      <option
        value="08"
      >
        August
      </option>
      <option
        value="09"
      >
        September
      </option>
      <option
        value="10"
      >
        October
      </option>
      <option
        value="11"
      >
        November
      </option>
      <option
        value="12"
      >
        December
      </option>
    </select>
    <div>
      <label
        htmlFor="category"
      >
        Category 
      </label>
      <select
        name="category"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select Category 
        </option>
      </select>
    </div>
    <section>
      <TransactionTable
        headings={
          Array [
            "Date",
            "Title",
            "Amount",
            "Category",
          ]
        }
        rows={Array []}
      />
    </section>
    <Link
      to="entry"
    >
      <button>
        New Transaction
      </button>
    </Link>
  </main>
</div>
`;
